tools/mac-distribution/packaging/appbundle.go

```diff
@@ -0,0 +1,112 @@
+package packaging
+
+import (
+	"fmt"
+	"log/slog"
+	"os"
+	"path/filepath"
+
+	"github.com/gravitational/shared-workflows/tools/mac-distribution/internal/fileutil"
+	"github.com/gravitational/shared-workflows/tools/mac-distribution/notarize"
+	"github.com/gravitational/trace"
+)
+
+// AppBundlePackager is a packager for creating an app bundle (.app) for distribution.
+type AppBundlePackager struct {
+	Info *AppBundleInfo
+
+	log        *slog.Logger
+	notaryTool *notarize.Tool
+}
+
+// AppBundleInfo represents an app bundle to be packaged for distribution.
+type AppBundleInfo struct {
+	// Skeleton directory to use as the base for the app bundle
+	Skeleton string
+	// Entitlements file to use for the app bundle
+	Entitlements string
+	// AppBinary is the binary to use as the main executable for the app bundle
+	AppBinary string
+}
+
+// AppBundlePackagerOpts contains options for creating an AppBundlePackager.
+type AppBundlePackagerOpts struct {
+	// NotarizationEnabled determines whether to notarize the app bundle
+	NotaryTool *notarize.Tool
+	// Logger is the logger to use for the packager
+	Logger *slog.Logger
+}
+
+// NewAppBundlePackager creates a new AppBundlePackager.
+func NewAppBundlePackager(info *AppBundleInfo, opts *AppBundlePackagerOpts) *AppBundlePackager {
+	log := opts.Logger
+	if log == nil {
+		log = slog.Default()
+	}
+
+	return &AppBundlePackager{
+		Info:       info,
+		log:        log,
+		notaryTool: opts.NotaryTool,
+	}
+}
+
+// Package creates an app bundle from the provided info.
+func (a *AppBundlePackager) Package() error {
+	if err := a.Info.validate(); err != nil {
+		return trace.Wrap(err)
+	}
+
+	// Copy the app binary into the skeleton
+	binDir := filepath.Join(a.Info.Skeleton, "Contents", "MacOS")
+	if err := os.MkdirAll(binDir, 0755); err != nil {
+		return trace.Wrap(err)
+	}
+
+	binDest := filepath.Join(binDir, filepath.Base(a.Info.AppBinary))
+	if err := fileutil.CopyFile(a.Info.AppBinary, binDest); err != nil {
+		return trace.Wrap(err)
+	}
+
+	if a.notaryTool == nil {
+		a.log.Info("notarization skipped")
+		return nil
+	}
+
+	// Notarize the app bundle
+	if err := a.notaryTool.NotarizeAppBundle(a.Info.Skeleton, notarize.AppBundleOpts{Entitlements: a.Info.Entitlements}); err != nil {
+		return trace.Wrap(err)
+	}
+
+	return nil
+}
+
+func (a *AppBundleInfo) validate() error {
+	// Validate skeleton
+	info, err := os.Stat(a.Skeleton)
+	if err != nil {
+		return trace.Wrap(err, "skeleton directory does not exist")
+	}
+
+	if !info.IsDir() {
+		return fmt.Errorf("skeleton must be a directory")
```

Comment:
```suggestion
		return errors.New("skeleton must be a directory")
```

Use `errors.New` if you just have a string literal. The `fmt` version is only necessary when you have format verbs that you want to do substitution with.

Reply:
Since the `gravitational/trace` library is being used in this module, I'd suggest this should be `trace.Errorf()` here and all places that errors are created other than wrapped.